src/vehicle.cpp

```diff
@@ -3113,28 +3113,32 @@ float vehicle::strain ()
         return (float) (abs(velocity) - sv) / (float) (mv - sv);
 }
 
-bool vehicle::valid_wheel_config ()
+bool vehicle::sufficient_wheel_config ()
 {
     std::vector<int> floats = all_parts_with_feature(VPFLAG_FLOATS);
     if( !floats.empty() ) {
         return floats.size() > 2;
     }
-
-    int x1 = 0, y1 = 0, x2 = 0, y2 = 0;
-    int count = 0;
     std::vector<int> wheel_indices = all_parts_with_feature(VPFLAG_WHEEL);
     if(wheel_indices.empty()) {
-        //No wheels!
+        // No wheels!
         return false;
     } else if(wheel_indices.size() == 1) {
-        //Has to be a stable wheel
-        if(part_info(wheel_indices[0]).has_flag("STABLE")) {
-            //Valid only if the vehicle is 1 square in size (1 structural part)
-            return (all_parts_at_location(part_location_structure).size() == 1);
-        } else {
+        // Has to be a stable single wheel
+        if(!part_info(wheel_indices[0]).has_flag("STABLE")) {
```

Comment:
@BevapDin 
`return part_info(...)` is only equivalent right now because this is the last condition. if a third condition is added later, it would come between this `return false` and the final `return true`.

The reason to split this into two functions is for the changes in veh_interact.cpp above, so that the vehicle examine screen can differentiate between having too few wheels and having an imbalanced vehicle. I think that's a useful interface change even if my mechanic changes aren't accepted here.
https://github.com/CleverRaven/Cataclysm-DDA/pull/11691/files#diff-0c2198543dfd6abf16ced57ef86b794cR1538